doc/shared/suncontrol/SUNControl_Description.rst

```diff
@@ -0,0 +1,480 @@
+
+Thus *dsm* values below one represent errors estimated to be more accurate than
+needed, whereas errors above one are considered to be larger than allowable.
+
+The base ``SUNControl`` class is modeled after SUNDIALS' other object-oriented
+classes, in that this class contains a pointer to an implementation-specific
+*content*, an *ops* structure with generic controller operations, and a
+:c:type:`SUNContext` object. Specifically, the type ``SUNControl`` is defined
+as:
+
+.. c:type:: struct _generic_SUNControl *SUNControl
+
+and the base class structure is defined as
+
+.. code-block:: C
+
+   struct _generic_SUNControl {
+        void* content;
+        _generic_SUNControl_Ops* ops;
+        SUNContext sunctx;
+    };
+
+Here, ``_generic_SUNControl_Ops`` is the pointer to a structure containing
+function pointers to the various controller operations, and is defined as
+
+.. code-block:: c
+
+    struct _generic_SUNControl_Ops {
+        SUNControl_ID (*getid)(SUNControl C);
+        void          (*destroy)(SUNControl C);
+        int           (*estimatestep)(SUNControl C, realtype h, realtype dsm, realtype* hnew);
+        int           (*estimatestepandorder)(SUNControl C, realtype h, int q, realtype dsm, realtype* hnew, int *qnew);
+        int           (*estimatemristeps)(SUNControl C, realtype H, realtype h, realtype DSM, realtype dsm, realtype* Hnew, realtype *hnew);
+        int           (*estimatesteptol)(SUNControl C, realtype H, realtype tolfac, realtype DSM, realtype dsm, realtype *Hnew, realtype* tolfacnew);
+        int           (*reset)(SUNControl C);
+        int           (*setdefaults)(SUNControl C);
+        int           (*write)(SUNControl C, FILE* fptr);
+        int           (*setmethodorder)(SUNControl C, int q);
+        int           (*setembeddingorder)(SUNControl C, int p);
+        int           (*seterrorbias)(SUNControl C, realtype bias);
+        int           (*update)(SUNControl C, realtype h, realtype dsm);
+        int           (*updatemrih)(SUNControl C, realtype H, realtype h, realtype DSM, realtype dsm);
+        int           (*updatemritol)(SUNControl C, realtype H, realtype tolfac, realtype DSM, realtype dsm);
+        int           (*space)(SUNControl C, long int *lenrw, long int *leniw);
+    };
+
+
+
+SUNControl IDs
+--------------
+
+The time integrators in SUNDIALS adapt a variety of parameters to achieve
+accurate and efficient computations. To this end, each SUNControl implementation
+should note its type, so that integrators will understand the types of
+adaptivity that the controller is designed to perform. These are encoded in the
+following set of SUNControl types:
+
+.. _SUNControl.Description.controllerIDs:
+.. table:: Identifiers associated with SUNControl implementations
+   :align: center
+
+   ========================  =====================================================
+   SUNControl ID             Intended form of adaptivity
```

Comment:
> Some more naming convention questions, should this be `SUNControl_Type` or `SUNControl_Type` and `SUNDIALS_CONTROL_NONE` or `SUN_CONTROL_NONE`?

The naming convention we defined implies it could be either, however I prefer `SUN_ADAPTCONTROLLER_NONE` and `SUNAdaptController_Type` (another comment chain below explains why `SUNAdaptController`.